### PR TITLE
When tls is on, use plain framer and handshaker to avoid double encryption

### PR DESCRIPTION
**Pull request by perusworld**
When tls is on, use passthrough framer and passthrough handshaker to avoid double encryption

Signed-off-by: Saravana Perumal Shanmugam <perusworld@linux.com>

### CODE REVIEW

**Comment by atoulme** [comment]
By passthrough, do you mean unencrypted? Would you please look at renaming passthrough to “plain”?

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/plain/PlainFramer.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import io.netty.buffer.ByteBuf;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class PlainFramer extends Framer {
+  private static final Logger LOG = LogManager.getLogger();
+
+  public PlainFramer() {
+    LOG.trace("Init");
```
Comment (lucassaldanha) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/plain/PlainFramer.java:32]: NIT suggestion
```
LOG.trace("Initialising PlainFramer");
```

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/plain/PlainFramer.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  @Override
+  public synchronized MessageData deframe(final ByteBuf buf) throws FramingException {
+    LOG.trace("deframe");
```
Comment (lucassaldanha) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/plain/PlainFramer.java:37]: NIT suggestion
```
LOG.trace("Deframing message");
```

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/plain/PlainFramer.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    }
+    PlainMessage message = MessageHandler.parseMessage(buf);
+    checkState(
+        MessageType.DATA.equals(message.getMessageType()), "unexpected message: needs to be data");
+    return new RawMessage(message.getCode(), message.getData());
+  }
+
+  @Override
+  public synchronized void frame(final MessageData message, final ByteBuf output) {
+    LOG.trace("frame");
```
Comment (lucassaldanha) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/plain/PlainFramer.java:49]: NIT suggestion
```
LOG.trace("Framing message");
```

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/plain/PlainMessage.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  private final MessageType messageType;
+  private final int code;
+  private final Bytes data;
+
+  public PlainMessage(final MessageType messageType, final byte[] data) {
+    super();
+    this.messageType = messageType;
+    this.code = -1;
+    this.data = Bytes.wrap(data);
+  }
```
Comment (lucassaldanha) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/plain/PlainMessage.java:30]:
```suggestion
  public PlainMessage(final MessageType messageType, final byte[] data) {
    this(messageType, Bytes.wrap(data));
  }
```

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/plain/PlainMessage.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+  public PlainMessage(final MessageType messageType, final Bytes data) {
+    super();
+    this.messageType = messageType;
+    this.code = -1;
+    this.data = data;
+  }
```
Comment (lucassaldanha) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/plain/PlainMessage.java:37]:
```suggestion
  public PlainMessage(final MessageType messageType, final Bytes data) {
    this(messageType, -1, data);
  }
```

**Review by lucassaldanha — CHANGES_REQUESTED** [review]
PR is looking good. However, there are a few things that we need to fix:
1. All **new** files need to have the updated license header (example: https://github.com/hyperledger/besu/blob/main/consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/validator/ValidatorModeTransitionLogger.java)
2. Could you please add a changelog entry?

I have also added a few minor comments but the main reason for requesting changes is the license header issue.

**Comment by atoulme** [comment]
Thank you for addressing my feedback! PR intent looks great to me.

**Review by lucassaldanha — APPROVED** [review]

**Pull request merged**